### PR TITLE
12520 edit url fix

### PR DESCRIPTION
**Pull request by thomasmoors**

### CODE REVIEW

File: includes/MediaBynderStreamWrapper.inc [hunk 1]
```diff
@@ -9,17 +9,17 @@
...
-        $this->base_url = variable_get('media_bynder_url', '');
-
-        if ($parameters = $this->get_parameters()) {
-            return media_bynder_embed_url($parameters['i']);
-        } else {
-            $parameters = media_bynder_uri_parse_style_url($this->uri);
-            if($parameters){
-                return media_bynder_embed_url($parameters['i'], $parameters['style'], $parameters['effects']);
+        $parts = explode('/', $this->uri);
+
```
Comment (pubkraal) [inline includes/MediaBynderStreamWrapper.inc:13]: some extra unnecessary whitespace!

File: includes/MediaBynderStreamWrapper.inc [hunk 2]
```diff
@@ -9,17 +9,17 @@
...
-        } else {
-            $parameters = media_bynder_uri_parse_style_url($this->uri);
-            if($parameters){
-                return media_bynder_embed_url($parameters['i'], $parameters['style'], $parameters['effects']);
+        $parts = explode('/', $this->uri);
+      
+        foreach ($parts as $part => $part_key) {
+            if ($part_key == 'i'){
+                //idHash of URI
+                if (count($parts) >= ($part + 1)) {
```
Comment (pubkraal) [inline includes/MediaBynderStreamWrapper.inc:17]: This looks a bit finnicky, maybe there's a more elegant way of dealing with the parts? To be honest I don't even know what "i" means in part_key and that's why I'm lost.

Can you maybe refactor the code so it's more clear what this is trying to achieve?

File: includes/media_bynder.admin.inc [hunk 3]
```diff
@@ -15,6 +15,13 @@ include_once 'BynderRestClient.inc';
 function media_bynder_add($form, &$form_state = array(), $redirect = NULL) {
     module_load_include('inc', 'media', 'includes/media.browser');
 
+    $rest_client = media_bynder_rest_client();
+    $all_results = $rest_client->getFacets('');
+    if(is_string($all_results)) {
```
Comment (pubkraal) [inline includes/media_bynder.admin.inc:20]: please mind the code style :-).

File: includes/media_bynder.utils.inc [hunk 4]
```diff
@@ -25,52 +25,30 @@ function media_bynder_rest_client() {
...
-function media_bynder_embed_url($fragment_id, $type=false, $effects=false){
-    if(!$type){
-        $type = "web";
-    }
-    $base_url = variable_get('media_bynder_url', '');
-    if (substr($base_url, -1) != '/') {
-        $base_url .= '/';
-    }
-    $base_url .= 'm/' . $fragment_id . '/' . $type . '-drupal.jpg';
+function media_bynder_embed_url($fragment_id, $type=false, $effects=false, $name = null){
```
Comment (pubkraal) [inline includes/media_bynder.utils.inc:28]: Please remain consistent with the code-style!

File: includes/MediaBynderStreamWrapper.inc (hunk 2 above)
Comment (thomasmoors) [inline includes/MediaBynderStreamWrapper.inc:17]: the URI is supposed to have this format: bynder://f/<id>/i/<idHash>
by splitting (exploding) we extract the idHash and use it
it should always be the one after the /i/ segment. It checks if the array resulting from the split is long enough to store one segment after /i/, if not the fallback is to return FALSE.

File: includes/MediaBynderStreamWrapper.inc (hunk 1 above)
Comment (thomasmoors) [inline includes/MediaBynderStreamWrapper.inc:13]: fixed
